app/src/routes/api/v2/area.router.js

```diff
@@ -0,0 +1,305 @@
+        const userId = ctx.state.loggedUser.id;
+        let team = null;
+        try {
+            team = await TeamService.getTeamByUserId(userId);
+        } catch (e) {
+            logger.error(e);
+            ctx.throw(500, 'Team retrieval failed.');
+        }
+        if (team && team.areas.includes(ctx.params.id)) {
+            const areas = team.areas.filter(area => area !== ctx.params.id);
+            try {
+                await TeamService.patchTeamById(team.id, { areas });
+            } catch (e) {
+                logger.error(e);
+                ctx.throw(500, 'Team patch failed.');
+            }
+        }
+        const result = await AreaModel.remove({ _id: ctx.params.id });
+        if (!result || result.ok === 0) {
+            ctx.throw(404, 'Area not found');
+            return;
+        }
+        ctx.body = '';
+        ctx.statusCode = 204;
+    }
+
+    static async getAlertsOfArea(ctx){
+        logger.info(`Obtaining alerts of area with id ${ctx.params.id}`);
+        ctx.assert(ctx.query.precissionPoints, 400, 'precissionPoints is required');
+        ctx.assert(ctx.query.precissionBBOX, 400, 'precissionBBOX is required');
+        const result = await AreaModel.findOne({ _id: ctx.params.id });
+        if (!result) {
+            ctx.throw(404, 'Area not found');
+            return;
+        }
+        let generateImages = true;
+        if (ctx.query.nogenerate) {
+            generateImages = false;
+        }
+
+        let response = await AlertsService.groupAlerts(result, ctx.query.precissionPoints, ctx.query.precissionBBOX, generateImages);
+        ctx.body = response;
+    }
+
+}
+
+
+async function loggedUserToState(ctx, next) {
+    if (ctx.query && ctx.query.loggedUser){
+        ctx.state.loggedUser = JSON.parse(ctx.query.loggedUser);
+        delete ctx.query.loggedUser;
+    } else if (ctx.request.body && ctx.request.body.loggedUser) {
+        ctx.state.loggedUser = ctx.request.body.loggedUser;
+        delete ctx.request.body.loggedUser;
+    } else if (ctx.request.body.fields && ctx.request.body.fields.loggedUser) {
+        ctx.state.loggedUser = JSON.parse(ctx.request.body.fields.loggedUser);
+        delete ctx.request.body.loggedUser;
+    } else {
+        ctx.throw(401, 'Not logged');
+        return;
+    }
+    await next();
+}
+
+async function isMicroservice(ctx, next) {
+    if (ctx.state.loggedUser.id !== 'microservice'){
+        ctx.throw(403, 'Not authorized');
+    }
+    await next();
+}
+
+async function checkPermission(ctx, next) {
+    ctx.assert(ctx.params.id, 400, 'Id required');
+    let area = await AreaModel.findById(ctx.params.id);
+    if (!area) {
+        ctx.throw(404, 'Area not found');
+        return;
+    }
+    if (area.userId !== ctx.state.loggedUser.id && area.userId !== ctx.request.body.userId) {
+        ctx.throw(403, 'Not authorized');
+        return;
+    }
+    await next();
+}
+
+router.get('/', loggedUserToState, AreaRouterV2.getAll);
```

Comment:
Do we not want to check permissions on GET all?

Reply:
Depends on privacy and whether there is a use-case in the front-end for fetch all areas.

This GET method is more like "get all of _my_ AOIs" for a particular user. I think if you have an admin OATH then that _my_ areas is everything. Will have to test in staging.